src/sentry/utils/snuba.py

```diff
@@ -630,38 +635,50 @@ def raw_query(
         is_grouprelease=is_grouprelease,
         **kwargs,
     )
-    return bulk_raw_query([snuba_params], referrer=referrer)[0]
+    return bulk_raw_query([snuba_params], referrer=referrer, use_cache=use_cache)[0]
 
 
-def bulk_raw_query(snuba_param_list, referrer=None):
+def bulk_raw_query(snuba_param_list, referrer=None, use_cache=False):
     headers = {}
     if referrer:
         headers["referer"] = referrer
 
-    query_param_list = map(_prepare_query_params, snuba_param_list)
+    # Store the original position of the query so that we can maintain the order
+    query_param_list = list(enumerate(map(_prepare_query_params, snuba_param_list)))
 
-    def snuba_query(params):
-        query_params, forward, reverse, thread_hub = params
-        try:
-            with timer("snuba_query"):
-                referrer = headers.get("referer", "<unknown>")
-                if SNUBA_INFO:
-                    # We want debug in the body, but not in the logger, so dump the json twice
-                    logger.info(f"{referrer}.body: {json.dumps(query_params)}")
-                    query_params["debug"] = True
-                body = json.dumps(query_params)
-                with thread_hub.start_span(op="snuba", description=f"query {referrer}") as span:
-                    span.set_tag("referrer", referrer)
-                    for param_key, param_data in query_params.items():
-                        span.set_data(param_key, param_data)
-                    return (
-                        _snuba_pool.urlopen("POST", "/query", body=body, headers=headers),
-                        forward,
-                        reverse,
-                    )
-        except urllib3.exceptions.HTTPError as err:
-            raise SnubaError(err)
+    results = []
+
+    if use_cache:
+        # sqc - Snuba Query Cache
+        cache_keys = [
+            f"sqc:{sha1(json.dumps(query_params[0], sort_keys=True).encode('utf-8')).hexdigest()}"
```

Comment:
This includes the from_date and to_date, which I assume is intended. How often do the dates update?

Reply:
At this level, if you passed a date directly it'd be the exact date and the cache likely wouldn't be helpful. But at higher levels, we align the dates to larger buckets that are 1+ minutes in size, and so they're more stable.

Reply:
In discover we took to shifting time into 5 min windows so we could get better cache utilization. I'm not sure that is something we want to do universally, but queries could use that same approach as well.